core/src/main/java/com/radixdlt/rev2/REv2StateComputer.java

```diff
@@ -323,25 +323,19 @@ public LedgerProofBundle commit(LedgerExtension ledgerExtension, VertexStoreStat
           "Initial protocol update triggers must happen at epoch boundary");
     }
 
-    // A loop here allows us to enact protocol updates back-to-back
-    LedgerProof lastHandledProof;
     Option<com.radixdlt.statecomputer.commit.NextEpoch> maybeNextEpoch = none();
-    do {
-      lastHandledProof = newLatestProof;
-      final var newHeader = newLatestProof.ledgerHeader();
-      if (newHeader.nextEpoch().isPresent()) {
-        latestProofWhichInitiatedAnEpochChange = newLatestProof;
-        maybeNextEpoch = newHeader.nextEpoch();
-      }
-      if (newHeader.nextProtocolVersion().isPresent()) {
-        final var nextProtocolVersion = newHeader.nextProtocolVersion().unwrap();
-        latestProofWhichInitiatedAProtocolUpdate = Option.some(newLatestProof);
-        final var finalExecutionProof =
-            this.rustProtocolUpdate.applyProtocolUpdate(nextProtocolVersion).postUpdateProof();
-        newLatestProof = finalExecutionProof;
-        latestProtocolUpdateExecutionProof = Option.some(finalExecutionProof);
-      }
-    } while (newLatestProof != lastHandledProof);
+    final var newHeader = newLatestProof.ledgerHeader();
```

Comment:
I previously created this loop in the last PR, but decided it made more sense handling this loop in rust in one place - which is now triggered by `applyKnownPendingProtocolUpdate`.